tools/checkdocs/checkdocs.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"io"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+)
+
+type FileSlug struct {
+	slug     string
+	fileName string
+}
+
+func main() {
+	dir := flag.String("i", "website/blog", "root of website blog directory")
+
+	fs, err := os.ReadDir(*dir)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	slugs := GetBlogSlugs(fs)
+	for _, slug := range slugs {
+		fo, err := os.Open(filepath.Join(*dir, slug.fileName))
```

Comment:
```suggestion
		fo, err := os.Open(filepath.Join(*dir, slug.fileName))
		defer fo.Close()
```

Let's make sure to close the file, defer closes it when this function returns.